docs/usage.rst

```diff
@@ -69,6 +69,82 @@ callable is called, or to prevent it, use the ``call_default`` parameter:
     Note that ``call_default`` is only used when the related setting is missing
     from the project settings!
 
+Inherit from environment variables
+''''''''''''''''''''''''''''''''''
+
+Nowadays it became more and more popular to inherit settings from the environment.
+This functionality is supported as well. If the setting name is found in ``os.environ``,
+this value is used. To distinguish between different setting types, the value is
+parsed as a JSON by default. Some specific setting types support even other, easier
+and nicer, formats. For more information see the example below.
+
+Anyway, you can override the environ value parsing on your own. It is done by the
+``decode_environ(self, value)`` method.
+
+Example
+-------
+
+Let's assume following setting class, following django setting module and following
+illustration file:
+
+.. code:: python
+
+    # setting class
+    class MySettings(AppSettings):
+        example = StringSetting()
+        array = IterableSetting()
+        dictionary = DictSetting()
+        boolean = BooleanSetting()
+
+    # DJANGO_SETTING_MODULE
+    SECRET_KEY = "__MY_SUPER_SECRET__"  # required setting by Django itself
+    EXAMPLE = "example"
+    ARRAY = [1, 2, 3]
+    DICTIONARY = {"key": "value"}
+    BOOLEAN = True
+
+    # Illustration file
+    SETTINGS = MySettings()
+    SETTINGS.check()
+
+    print(SETTINGS.example)
+    print(SETTINGS.array)
+    print(SETTINGS.dictionary)
+    print(SETTINGS.boolean)
+
+Let's see, what we get:
+
+.. code::
+
+    $ python illustration.py
+    > example
+    > [1, 2, 3]
+    > {"key": "value"}
+    > True
+
+    $ EXAMPLE='"environ"' ARRAY='[7, 8, 9]' DICTIONARY='{"env": "iron"}' BOOLEAN='false' python illustration.py
+    > environ
+    > [7, 8, 9]
+    > {"env": "iron"}
+    > False
+
+``ListSetting`` and ``DictSetting`` support even other formats of environ values by default.
```

Comment:
I think it's time to add a documentation of available `Setting` objects, where the structure of environment values could be described.

Reply:
Or alternatively describe it in class docstring which will be displayed in autogenerated reference page.

Reply:
Should it be id the *class* docstring or the decode_environ method docstring is sufficient enough?

Reply:
I'd go with the class docstring.